cli/src/main/java/org/apache/falcon/cli/FalconEntityCLI.java

```diff
@@ -259,17 +288,21 @@ public void entityCommand(CommandLine commandLine, FalconClient client) throws F
             validateColo(optionsList);
             validateNotEmpty(entityName, FalconCLIConstants.ENTITY_NAME_OPT);
             result = client.update(entityType, entityName, filePath, skipDryRun, doAsUser).getMessage();
+        } else if (optionsList.contains(SUBMIT_AND_SCHEDULE_OPT)) {
+            validateNotEmpty(filePath, FILE_PATH_OPT);
+            validateColo(optionsList);
+            result = client.submitAndSchedule(entityType, filePath, skipDryRun, doAsUser, userProps).getMessage();
         }  else if (optionsList.contains(FalconCLIConstants.UPDATE_CLUSTER_DEPENDENTS_OPT)) {
             validateNotEmpty(cluster, FalconCLIConstants.CLUSTER_OPT);
             result = client.updateClusterDependents(cluster, skipDryRun, doAsUser).getMessage();
+        } else if (optionsList.contains(VALIDATE_OPT)) {
+            validateNotEmpty(filePath, FILE_PATH_OPT);
+            validateColo(optionsList);
+            result = client.validate(entityType, filePath, skipDryRun, doAsUser).getMessage();
         } else if (optionsList.contains(FalconCLIConstants.SUBMIT_AND_SCHEDULE_OPT)) {
```

Comment:
Not true. not empty and colo validations are being done in validation of other options too. I agree that the middle two lines are same, but abstracting just two lines out in a method would be over-engineering.